### PR TITLE
Split aws_iam_role_policy_attachment

### LINKED ISSUE #724: Split iam_role_policy_attachment_supplier

### PR DESCRIPTION
**Pull request by eliecharra**
Fix #724

### CODE REVIEW

**Review by lotoussa — APPROVED** [review]
lgtm

File: pkg/remote/test/aws_iam_role_policy_attachment_multiple/results.golden.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
+[
```
Comment (lotoussa) [inline pkg/remote/test/aws_iam_role_policy_attachment_multiple/results.golden.json:1]: other PR have not the aws_... naming but only iam_role_... for the directory

File: pkg/remote/aws/iam_role_policy_attachment_details_fetcher.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+func NewIamRolePolicyAttachmentDetailsFetcher(provider terraform.ResourceReader, deserializer *resource.Deserializer) *IamRolePolicyAttachmentDetailsFetcher {
+	return &IamRolePolicyAttachmentDetailsFetcher{
+		reader:       provider,
+		deserializer: deserializer,
+	}
+}
+
+func (r *IamRolePolicyAttachmentDetailsFetcher) ReadDetails(res resource.Resource) (resource.Resource, error) {
+	ctyVal, err := r.reader.ReadResource(terraform.ReadResourceArgs{
+		Ty: aws.AwsIamRolePolicyAttachmentResourceType,
```
Comment (sundowndev) [inline pkg/remote/aws/iam_role_policy_attachment_details_fetcher.go:23]: I know we already did that in every other details fetchers, but is there a reason to not use the resource type defined by the enumerator ? We could just call `res.TerraformType()` here

**Review by sundowndev — DISMISSED** [review]

**Review by lotoussa — APPROVED** [review]
rebase main, approving review.

**Pull request merged**